--- a/SimTKcommon/Geometry/src/DecorativeGeometry.cpp
+++ b/SimTKcommon/Geometry/src/DecorativeGeometry.cpp
@@ -34,17 +34,17 @@
 // Some common RGB values; these constants are global external symbols exported
 // by the library, defined as externs in DecorativeGeometry.h.
 
-const Vec3 Black   = Vec3( 0, 0, 0);
-const Vec3 Gray    = Vec3(.5,.5,.5);
-const Vec3 Red     = Vec3( 1, 0, 0);
-const Vec3 Green   = Vec3( 0, 1, 0);
-const Vec3 Blue    = Vec3( 0, 0, 1);
-const Vec3 Yellow  = Vec3( 1, 1, 0);
-const Vec3 Orange  = Vec3( 1,.5, 0);
-const Vec3 Magenta = Vec3( 1, 0, 1);
-const Vec3 Purple  = Vec3(.5, 0,.5);
-const Vec3 Cyan    = Vec3( 0, 1, 1);
-const Vec3 White   = Vec3( 1, 1, 1);
+constexpr Vec3 Black   = Vec3( 0, 0, 0);
+constexpr Vec3 Gray    = Vec3(.5,.5,.5);
+constexpr Vec3 Red     = Vec3( 1, 0, 0);
+constexpr Vec3 Green   = Vec3( 0, 1, 0);
+constexpr Vec3 Blue    = Vec3( 0, 0, 1);
+constexpr Vec3 Yellow  = Vec3( 1, 1, 0);
+constexpr Vec3 Orange  = Vec3( 1,.5, 0);
+constexpr Vec3 Magenta = Vec3( 1, 0, 1);
+constexpr Vec3 Purple  = Vec3(.5, 0,.5);
+constexpr Vec3 Cyan    = Vec3( 0, 1, 1);
+constexpr Vec3 White   = Vec3( 1, 1, 1);
 
     /////////////////////////
     // DECORATIVE GEOMETRY //
